### PR TITLE
Add Firewall Policy resource for VWAN implementation

### PR DESCRIPTION
**Pull request by krowlandson**
## Overview/Summary

Currently, only the Hub & Spoke implementation includes a Firewall Policy for the Azure Firewall deployment.

This PR adds creation of a Firewall Policy and associates it with the Azure Firewall deployed to a Virtual Hub.

I have used all existing input parameters, so no changes should be needed in the Portal experience.

## This PR fixes/adds/changes/removes

1. Add a `Microsoft.Network/firewallPolicies` resource and link it to the `Microsoft.Network/azureFirewalls` resource in the `eslzArm/subscriptionTemplates/vwan-connectivity.json` deployment template.
2. Add this update to the `docs/wiki/Whats-new.md` page.

### Breaking Changes

none

## Testing Evidence

image

image

image

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

> **NOTE**: I do not believe there are any Wiki docs which need updating with this change (other than the What's new page) but please let me know if this would be useful to add somewhere.

### CODE REVIEW

**Comment by jtracey93** [comment]
Will test ALZ Accelerator (portal) with URL https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fkrowlandson%2FEnterprise-Scale%2Fvwan-addfirewallpolicy%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fkrowlandson%2FEnterprise-Scale%2Fvwan-addfirewallpolicy%2FeslzArm%2Feslz-portal.json

**Comment by jtracey93** [comment]
> Will test ALZ Accelerator (portal) with URL https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fkrowlandson%2FEnterprise-Scale%2Fvwan-addfirewallpolicy%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fkrowlandson%2FEnterprise-Scale%2Fvwan-addfirewallpolicy%2FeslzArm%2Feslz-portal.json

This worked as expected 👍

image

image

image

However we should add DNS proxy support like we do in Hub & Spoke Model and advertise in the portal. Will create PR review for this

File: eslzArm/subscriptionTemplates/vwan-connectivity.json [hunk 1]
```diff
@@ -194,13 +198,27 @@
...
                                             }
                                         },
+                                        {
+                                            "condition": "[equals(parameters('enableAzFw'), 'Yes')]",
+                                            "type": "Microsoft.Network/firewallPolicies",
+                                            "apiVersion": "2020-11-01",
+                                            "name": "[variables('azFwPolicyName')]",
+                                            "location": "[parameters('location')]",
+                                            "properties": {
+                                                "dnsSettings": "[json('null')]",
```
Comment (jtracey93) [inline eslzArm/subscriptionTemplates/vwan-connectivity.json:208]:
```suggestion
                                                "dnsSettings": "[if(equals(parameters('enableAzFwDnsProxy'), 'Yes'), variables('azFirewallDnsSettings'), json('null'))]",
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]
All works as per comment https://github.com/Azure/Enterprise-Scale/pull/922#issuecomment-1043184913

However, we should add DNS proxy support to match traditional Hub & Spoke Module and also to meet what the portal option displays for VWAN deployment (shown below):

image

To resolve this in this PR I believe we need to do the following:

1. Add parameter for `enableAzFwDnsProxy` to `vwan-connectivity.json` 

```json
"enableAzFwDnsProxy": {
            "type": "string",
            "allowedValues": [
                "Yes",
                "No"
            ],
            "defaultValue": "No",
            "metadata": {
                "description": "Select whether the Azure Firewall should be used as DNS Proxy or not."
            }
        }
```

2. Add variable for `azFirewallDnsSettings` to `vwan-connectivity.json` 

```json
"azFirewallDnsSettings": {
            "enableProxy": true
        }
```
3. Change line 208 in `vwan-connectivity.json` for `dnsSettings` to the same as hub spoke model - https://github.com/Azure/Enterprise-Scale/blob/4134a349b2fbb26117f209d474c1cb4f319876ed/eslzArm/subscriptionTemplates/hubspoke-connectivity.json#L405 (suggestion via PR review made for this)
4. Add as input parameter in `eslzArm.json` for `enableAzFwDnsProxy` to `vwan-connectivity.json`  nested deployment - (in this block https://github.com/Azure/Enterprise-Scale/blob/4134a349b2fbb26117f209d474c1cb4f319876ed/eslzArm/eslzArm.json#L1735)
    - Use same input as hub spoke model - https://github.com/Azure/Enterprise-Scale/blob/4134a349b2fbb26117f209d474c1cb4f319876ed/eslzArm/eslzArm.json#L1613

```json
"enableAzFwDnsProxy": {
                        "value": "[parameters('enableAzFwDnsProxy')]"
                    }
```

This should then allow the DNS proxy option from the portal to be honoured as this is supported as seen: 
image

Great work here though @krowlandson 👍

**Comment by krowlandson** [comment]
## Successfully ran Portal deployment with updates requested to add DNS proxy settings:

image

## Connectivity settings to show DNS proxy enabled:

image

## Identity settings to show identity Subscription with spoke virtual network:

image

## DNS Proxy settings visible on Firewall Policy:

image

## DNS Proxy settings visible on spoke virtual network for identity:

image

**Review by jtracey93 — APPROVED** [review]
Nice work @krowlandson. 

LGTM 👍

**Pull request merged**